src/components/views/Waiter/Waiter.js

```diff
@@ -1,16 +1,166 @@
+import Paper from '@material-ui/core/Paper';
+import Button from '@material-ui/core/Button';
+
+class Waiter extends React.Component {
+  static propTypes = {
+    fetchTables: PropTypes.func,
+    loading: PropTypes.shape({
+      active: PropTypes.bool,
+      error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
+    }),
+    tables: PropTypes.array,
+    changeTableStatus: PropTypes.func,
+  };
+
+  componentDidMount() {
+    const { fetchTables } = this.props;
+    fetchTables();
+    // changeTableStatus();
+  }
+
+  renderActions(status, id, changeTableStatus) {
+    switch (status) {
+      case 'free':
+        return (
+          <>
+            <Button
+              onClick={() => {
+                changeTableStatus('thinking', id);
+              }}
+            >
+              thinking
+            </Button>
+            <Button
+              onClick={() => {
+                changeTableStatus('ordered', id);
+              }}
+            >
+              new order
+            </Button>
+          </>
+        );
+      case 'thinking':
+        return (
+          <Button
+            onClick={() => {
+              changeTableStatus('ordered', id);
+            }}
+          >
+            new order
+          </Button>
+        );
+      case 'ordered':
+        return (
+          <Button
+            onClick={() => {
+              changeTableStatus('prepared', id);
+            }}
+          >
+            prepared
+          </Button>
+        );
+      case 'prepared':
+        return (
+          <Button
+            onClick={() => {
+              changeTableStatus('delivered', id);
+            }}
+          >
+            delivered
+          </Button>
+        );
+      case 'delivered':
+        return (
+          <Button
+            onClick={() => {
+              changeTableStatus('paid', id);
+            }}
+          >
+            paid
+          </Button>
+        );
+      case 'paid':
+        return (
+          <Button
+            onClick={() => {
+              changeTableStatus('free', id);
+            }}
+          >
+            free
+          </Button>
+        );
+      default:
+        return null;
+    }
+  }
+
+  render() {
+    const {
+      loading: { active, error },
+      tables,
+      changeTableStatus,
+    } = this.props;
 
-function Waiter() {
-  return (
-    <div className={styles.component}>
-      <h2>Waiter view</h2>
-      <Link to={`${process.env.PUBLIC_URL}/waiter/order/new`}>New</Link>
-      <Link to={`${process.env.PUBLIC_URL}/waiter/order/123abc`}>123abc</Link>
-    </div>
-  );
+    if (active || !tables.length) {
+      return (
+        <Paper className={styles.component}>
+          <p>Loading...</p>
+        </Paper>
+      );
+    } else if (error) {
+      return (
+        <Paper className={styles.component}>
+          <p>Error! Details:</p>
+          <pre>{error}</pre>
+        </Paper>
+      );
+    } else {
+      return (
```

Comment:
```suggestion
    if (active ) { // jeśli naprawisz błąd ze złą strukturą `inistalState` to sprawdzanie czy tablica jest pusta nie będzie konieczny
      return (
        <Paper className={styles.component}>
          <p>Loading...</p>
        </Paper>
      );
    } 
    if (error) {  //nie potrzeby else -> jeśli tamten warunek bł prawdziwy to i tak tu nie wejdzie bo był już wczesniej return
      return (
        <Paper className={styles.component}>
          <p>Error! Details:</p>
          <pre>{error}</pre>
        </Paper>
      );
    } 
    return ( //to samo co wyżej
```

Dodatkowo możesz obsłużyć przypadek że jeśli tablica stołow jest pusta to wyświetlić info o braku danych

Reply:
Done.